### DISCUSSION TITLE
Search only certain properies

### DISCUSSION THREAD

**Discussion by Heavm00** [microsoft-search/pnp-modern-search#2339] [Q&A] [answered]
v3

I would like to implement a people search (search box, search results), where I just want to search certain properties (lastname, firstname), but search should not include e-mail or accountname. I modiefied my selected properties, but the search result webpart always automatically add the following properties: `Title,Path,OriginalPath,SiteLogo,contentclass,FileExtension,Filename,ServerRedirectedURL,DefaultEncodingURL,IsDocument,IsContainer,IsListItem,FileType,HtmlFileType,NormSiteID,NormWebID,NormListID,NormUniqueID,Created,PreviewUrl,PictureThumbnailURL,ServerRedirectedPreviewURL,HitHighlightedSummary,ServerRedirectedEmbedURL,ParentLink,owstaxidmetadataalltagsinfo,Author,AuthorOWSUSER,SPSiteUrl,SiteTitle,SiteId,WebId,UniqueID,AccountName`

In my case the e-mail adress/account name contains the name of persons  (eg. anyone@maier-consult.com). When I'm looking for 'maier' I just want to get people with lastname or firstname = maier. In my case I'm getting all users, because of their e-mail address/account name.  

I tried to create a custom managed property for workmail (not searchable). That property can be set in selected properties. But I cannot remove 'AccountName'.

Is there any way to achive that? Thanks.

**Comment by wobba** [comment, marked as answer]
Try using a query template like this: `FirstName:({searchTerms} OR LastName:({searchTerms}))`

**Reply by Heavm00** [reply]
Oh great - what a simple answer ;-) Thank you very much!
One other question: is it also possible to combine multiple search boxes with one result webpart? (eg. search box for firstname, second one for lastname)?

**Reply by wobba** [reply]
Sorry no, that is not possible with any version of these web parts. You would have to develop your own customized version to make this work.
